testplan/runners/pools/connection.py

```diff
@@ -2,46 +2,348 @@
+        self._sock = self._context.socket(zmq.REQ)
+        self._sock.connect("tcp://{}".format(self._address))
+        self.active = True
+
+    def disconnect(self):
+        """Disconnect from Server"""
+        self.active = True
+        self._sock.close()
+        self._sock = None
+        self._context.destroy()
+        self._context = None
+
+    def send(self, message):
+        """
+        Worker sends a message.
+
+        :param message: Message to be sent.
+        :type message: :py:class:`~testplan.runners.pools.communication.Message`
+        """
+        if self.active:
+            self._sock.send(pickle.dumps(message))
+
+    def receive(self):
+        """
+        Worker tries to receive the response to the message sent until timeout.
+
+        :return: Response to the message sent.
+        :type: :py:class:`~testplan.runners.pools.communication.Message`
+        """
+        start_time = time.time()
+
+        while self.active:
+            try:
+                received = self._sock.recv(flags=zmq.NOBLOCK)
+                try:
+                    loaded = pickle.loads(received)
+                except Exception as exc:
+                    print('Deserialization error. - {}'.format(exc))
+                    raise
+                else:
+                    return loaded
+            except zmq.Again:
+                if time.time() - start_time > self._recv_timeout:
+                    print('Transport receive timeout {}s reached!'.format(
+                        self._recv_timeout))
+                    return None
+                time.sleep(self._recv_sleep)
+        return None
+
+
+class ZMQClientProxy(object):
+    """
+    Representative of a process worker's transport in local worker object.
+    """
+
+    def __init__(self):
+        self.connection = None
+        self.address = None
+
+    def respond(self, message):
+        """
+        Used by :py:class:`~testplan.runners.pools.base.Pool` to respond to
+        worker request.
+
+        :param message: Respond message.
+        :type message: :py:class:`~testplan.runners.pools.communication.Message`
+        """
+        self.connection.send(pickle.dumps(message))
+
+
+@six.add_metaclass(abc.ABCMeta)
+class Server(entity.Resource):
+    """
+    Abstract base class for pools to communicate to its workers.
     """
 
     def __init__(self):
-        super(TCPConnectionManager, self).__init__()
+        super(Server, self).__init__()
+        self._workers = []
+
+    @property
+    def workers(self):
+        return self._workers
+
+    def starting(self):
+        """Server starting logic."""
+        self.status.change(self.status.STARTED)
+
+    def stopping(self):
+        """Server stopping logic."""
+
+        self._unregister_workers()
+        self.status.change(self.status.STOPPED)
+
+    def aborting(self):
+        """Abort policy - no abort actions are required in the base class."""
+        pass
+
+    @abc.abstractmethod
```

Comment:
I'm not sure that the register/unregister methods need to be marked as abstract - do they always need to be overridden? The base implementation seems useful.